syntax = "proto3";
package md;

message PingRequest { string message = 1; }
message PingReply   { string message = 1; }

message SubscribeRequest { string instrument_id = 1; }

message Snapshot {
  string instrument_id = 1;
  repeated double prices = 2;
}

message Incremental {
  string instrument_id = 1;
  int64 ts_ms = 2;
  double value = 3; 
}

message Update {
  oneof kind {
    Snapshot snapshot = 1;
    Incremental incremental = 2;
  }
}

service MarketData {
  rpc Ping (PingRequest) returns (PingReply);
  rpc StreamPrices (SubscribeRequest) returns (stream Update);
}
